### PR TITLE
Update operator-sdk to 1.0.1

### PR DESCRIPTION
**Pull request by pliurh**

### CODE REVIEW

**Comment by pliurh** [comment]
Some files are moved/recreated by this the new operator-sdk framework. Please refer to https://sdk.operatorframework.io/docs/building-operators/golang/migration/

File: bindata/manifests/operator-webhook/003-webhook.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ webhooks:
         namespace: {{.Namespace}}
         path: "/validating-custom-resource"
     rules:
-      - operations: [ "CREATE", "UPDATE", "DELETE" ]
```
Comment (zshi-redhat) [inline bindata/manifests/operator-webhook/003-webhook.yaml:36]: I think we still need the DELETE operation for sriovnetworknodepolicies.

File: bindata/manifests/plugins/sriov-cni.yaml [hunk 2]
```diff
@@ -24,6 +24,7 @@ spec:
     spec:
       nodeSelector:
         beta.kubernetes.io/os: linux
+        node-role.kubernetes.io/worker:
```
Comment (zshi-redhat) [inline bindata/manifests/plugins/sriov-cni.yaml:27]: Why this is added? is it related to the operator-sdk update?

File: bundle.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM scratch
```
Comment (zshi-redhat) [inline bundle.Dockerfile:1]: Shall we remove this file? since the generated `bundle` is not used.

File: bundle.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+LABEL operators.operatorframework.io.bundle.channels.v1=alpha
+LABEL operators.operatorframework.io.bundle.channel.default.v1=
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.0.1
+LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
+LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v2
+LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
+LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
+
+COPY bundle/manifests /manifests/
+COPY bundle/metadata /metadata/
```
Comment (zshi-redhat) [inline bundle.Dockerfile:16]: I got an extra line when running make bundle.
`COPY bundle/tests/scorecard /tests/scorecard/`

File: bundle/manifests/sriov-network-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+apiVersion: operators.coreos.com/v1alpha1
```
Comment (zshi-redhat) [inline bundle/manifests/sriov-network-operator.clusterserviceversion.yaml:1]: Shall we remove `/bundle` for now if it's not used?

File: pkg/apply/merge.go [hunk 6]
```diff
@@ -13,6 +13,7 @@ func MergeMetadataForUpdate(current, updated *uns.Unstructured) error {
 
 	mergeAnnotations(current, updated)
 	mergeLabels(current, updated)
+	updated.SetResourceVersion(current.GetResourceVersion())
```
Comment (zshi-redhat) [inline pkg/apply/merge.go:16]: Why this is needed?

File: pkg/utils/utils.go [hunk 7]
```diff
@@ -594,3 +595,17 @@ func generateRandomGuid() net.HardwareAddr {
 
 	return guid
 }
+
+// GetWatchNamespace returns the Namespace the operator should be watching for changes
```
Comment (zshi-redhat) [inline pkg/utils/utils.go:599]: `os.Getenv("NAMESPACE")` is used in several other files to get operator namespace, why need a separate function?

File: hack/update-codegen.sh [hunk 8]
```diff
@@ -3,7 +3,7 @@
 chmod +x vendor/k8s.io/code-generator/generate-groups.sh
 ./vendor/k8s.io/code-generator/generate-groups.sh client,lister,informer \
       github.com/openshift/sriov-network-operator/pkg/client \
-      github.com/openshift/sriov-network-operator/pkg/apis \
+      github.com/openshift/sriov-network-operator/apis \
```
[the remark this answers is not in the document]
Comment (pliurh) [reply hack/update-codegen.sh:6]: No we don't need this script anymore

File: .gitignore [hunk 9]
```diff
@@ -1,8 +1,10 @@
-<<<<<<< HEAD
-=======
 # Temporary Build Files
 build/_output
 build/_test
+bin/
+testbin/
+env
+.env
```
[the remark this answers is not in the document]
Comment (pliurh) [reply .gitignore:7]: This is to ignore the temporary file generated during debugging and testing.

File: Makefile [hunk 10]
```diff
@@ -22,92 +23,181 @@ PKGS=$(shell go list ./... | grep -v -E '/vendor/|/test|/examples')
...
+# Build the bundle image.
+.PHONY: bundle-build
+bundle-build:
+	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+
+deploy-setup: install
 	@EXCLUSIONS=() hack/deploy-setup.sh $(NAMESPACE)
 
 deploy-setup-k8s: export NAMESPACE=sriov-network-operator
 deploy-setup-k8s: export ENABLE_ADMISSION_CONTROLLER=false
```
[the remark this answers is not in the document]
Comment (pliurh) [reply Makefile:172]: WATCH_NAMESPACE is not used by the the operator code. So we can just ignore it.

File: bundle.Dockerfile (hunk 3 above)
Comment (pliurh) [reply bundle.Dockerfile:1]: This is a generated file. I don't really know what is it for. I think we can keep it for now. ART will not process this file.

File: bundle.Dockerfile (hunk 4 above)
Comment (pliurh) [reply bundle.Dockerfile:16]: This is a generated file. Let's keep it as it is until we figure out what is it for.

File: bundle/manifests/sriov-network-operator.clusterserviceversion.yaml (hunk 5 above)
Comment (pliurh) [reply bundle/manifests/sriov-network-operator.clusterserviceversion.yaml:1]: Let's keep it for now. I'm trying to figure out how to generate the bundle files automatically with the new operator-sdk.

File: pkg/apply/merge.go (hunk 6 above)
Comment (pliurh) [reply pkg/apply/merge.go:16]: During testing, I found some error messages during reconciliation. It complains fail to update the resource without the `resourceVersrion`.

File: Makefile [hunk 11]
```diff
@@ -22,92 +23,181 @@ PKGS=$(shell go list ./... | grep -v -E '/vendor/|/test|/examples')
...
+	rm -rf $$KUSTOMIZE_GEN_TMP_DIR ;\
+	}
+KUSTOMIZE=$(GOBIN)/kustomize
+else
+KUSTOMIZE=$(shell which kustomize)
+endif
+
+# Generate bundle manifests and metadata, then validate generated files.
+.PHONY: bundle
+bundle: manifests
```
[the remark this answers is not in the document]
Comment (pliurh) [reply Makefile:157]: I'm trying to fix it in another patch.

File: Makefile [hunk 12]
```diff
@@ -22,92 +23,181 @@ PKGS=$(shell go list ./... | grep -v -E '/vendor/|/test|/examples')
...
-	@echo "$(GOFMT_CHECK)"
-	@echo ""
-	@echo "For details, run: gofmt -d -s $(GOFMT_CHECK)"
-	@echo ""
-	@exit 1
-endif
-
-verify: verify-gofmt
-
 sync-manifests-%:
```
[the remark this answers is not in the document]
Comment (pliurh) [reply Makefile]: Removed.

File: pkg/utils/utils.go (hunk 7 above)
Comment (pliurh) [reply pkg/utils/utils.go]: Removed.

File: bindata/manifests/plugins/sriov-cni.yaml (hunk 2 above)
Comment (pliurh) [reply bindata/manifests/plugins/sriov-cni.yaml:27]: No. Just want to keep the default nodeSelector aligned with the configure daemon.

File: bindata/manifests/operator-webhook/003-webhook.yaml (hunk 1 above)
Comment (pliurh) [reply bindata/manifests/operator-webhook/003-webhook.yaml:36]: There was some error when trying to delete the non-default policy.  I cannot find the root cause, so remote the deletion check. We add this back after fix that issue.

**Comment by zshi-redhat** [comment]
we need to merge this structure change to unblock other feature inclusion.
we will fix the remaining issues in this PR in later patches. 

/lgtm

**Pull request merged**